src/App.js

```diff
@@ -1,85 +1,215 @@
-        <h1>todos</h1>
+class TodoApp extends React.Component {
+  state = {
+    todos: [],
+    filter: 'all',
+    onSelectAllTodos: true,
+  }
 
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
+  componentDidMount() {
+    const initialState = JSON.parse(localStorage.getItem('TODOs'));
+
+    this.setState({
+      ...initialState,
+    });
+  }
+
+  componentDidUpdate() {
+    localStorage.setItem('TODOs', JSON.stringify(this.state));
+  }
+
+  addTodo = (newTodo) => {
+    this.setState(prevState => ({ todos: [...prevState.todos, newTodo] }));
+  }
+
+  changeStatus = (id) => {
+    this.setState(prevState => ({
+      todos: prevState.todos.map((todo) => {
+        if (todo.id === id) {
+          return {
+            ...todo,
+            completed: !todo.completed,
+          };
+        }
+
+        return todo;
+      }),
+    }));
+  }
+
+  changeTitle = (id, newTitle) => {
+    if (newTitle.trim() !== '') {
+      this.setState(prevState => ({
+        todos: prevState.todos.map((todo) => {
+          if (todo.id === id) {
+            return {
+              ...todo,
+              title: newTitle,
+            };
+          }
+
+          return todo;
+        }),
+      }));
+    } else {
+      const removeTodoId = this.state.todos
+        .findIndex(todo => todo.id === id);
+
+      this.setState((prevState) => {
+        const remainingTodos = [...prevState.todos];
+
+        remainingTodos.splice(removeTodoId, 1);
+
+        return (
+          { todos: [...remainingTodos] }
+        );
+      });
+    }
+  }
+
+  selectAllTodo = () => {
+    this.setState(prevState => ({
+      onSelectAllTodos: !prevState.onSelectAllTodos,
+      todos: prevState.todos.map(todo => ({
+        ...todo,
+        completed: prevState.onSelectAllTodos,
+      })),
+    }));
+  }
+
+  clearCompleted = () => {
+    this.setState(prevState => ({
+      todos: prevState.todos.filter(todo => todo.completed === false),
+    }));
+  }
+
+  filterViewAll = () => {
+    this.setState({
+      filter: 'all',
+    });
+  }
+
+  filterActive = () => {
+    this.setState({
+      filter: 'active',
+    });
+  }
+
+  filterCompleted = () => {
+    this.setState({
+      filter: 'completed',
+    });
+  }
+
+  deleteTodo = (event) => {
+    const removeTodoId = this.state.todos
+      .findIndex(item => item.id === +event.target.id);
+
+    this.setState((prevState) => {
+      const remainingTodos = [...prevState.todos];
+
+      remainingTodos.splice(removeTodoId, 1);
+
+      return (
+        { todos: [...remainingTodos] }
+      );
+    });
+  }
+
+  render() {
+    const {
+      todos,
+      filter,
+    } = this.state;
+
+    const notComplitedTodo = todos.filter(todo => todo.completed === false);
+    let preparedTodos;
+
+    if (filter === 'all') {
+      preparedTodos = [...todos];
+    }
+
+    if (filter === 'active') {
+      preparedTodos = [...todos].filter(todo => todo.completed === false);
```

Comment:
```suggestion
      preparedTodos = [...todos].filter(todo => !todo.completed);
```
there is no need to write boolean explicitly in this case